src/invert/laplace/invert_laplace.cxx

```diff
@@ -71,45 +71,56 @@ Laplacian::Laplacian(Options *options, const CELL_LOC loc, Mesh *mesh_in)
   // Communication option. Controls if asyncronous sends are used
   async_send = (*options)["async"].doc("Use asyncronous MPI send?").withDefault(true);
 
-  BoutReal filter = (*options)["filter"]
-                        .doc("Fraction of Z modes to filter out. Between 0 and 1")
-                        .withDefault(0.0);
-  int ncz = localmesh->LocalNz;
+  const BoutReal filter = (*options)["filter"]
+                              .doc("Fraction of Z modes to filter out. Between 0 and 1")
+                              .withDefault(0.0);
+  const int ncz = localmesh->LocalNz;
   // convert filtering into an integer number of modes
-  maxmode = ROUND((1.0 - filter) * static_cast<BoutReal>(ncz / 2));
-  // Can be overriden by max_mode option
-  OPTION(options, maxmode, maxmode);
-  if(maxmode < 0) maxmode = 0;
-  if(maxmode > ncz/2) maxmode = ncz/2;
+  maxmode = (*options)["maxmode"]
+                .doc("The maximum Z mode to solve for")
+                .withDefault(ROUND((1.0 - filter) * static_cast<BoutReal>(ncz / 2)));
 
-  OPTION(options, low_mem, false);
+  // Clamp maxmode between 0 and nz/2
+  if (maxmode < 0) {
+    maxmode = 0;
+  }
+  if (maxmode > ncz / 2) {
+    maxmode = ncz / 2;
+  }
+
+  low_mem = (*options)["low_mem"]
+                .doc("If true, reduce the amount of memory used")
+                .withDefault(false);
 
   nonuniform = (*options)["nonuniform"]
                    .doc("Use non-uniform grid corrections? Default is the mesh setting.")
                    .withDefault(coords->non_uniform);
 
-  all_terms = (*options)["all_terms"].doc("Include first derivative terms?").withDefault(true);
+  all_terms =
+      (*options)["all_terms"].doc("Include first derivative terms?").withDefault(true);
 
-  if (options->isSet("flags")) {
-    if ( options->isSet("global_flags") || options->isSet("inner_boundary_flags") || options->isSet("outer_boundary_flags") ) {
-      throw BoutException("Should not use old flags as well as new global_flags/inner_boundary_flags/outer_boundary_flags");
-    }
-    int flags = (*options)["flags"]
-                    .doc("Flags to control inner and outer boundaries.")
-                    .withDefault(0);
```

Comment:
I agree with removing this option in v5, but it might be a good idea to put a deprecation warning on use of the `laplace:flags` option in v4.3.1/v4.4. I can't think of anything better than printing a message to `output_warn` here, which isn't very noticable, but better than nothing...

Reply:
Good idea! I think there's a few options we'd like to deprecate like this too, although none spring to mind right now.